packages/@expo/cli/src/start/server/middleware/__tests__/ExpoGoManifestHandlerMiddleware-test.ts

```diff
@@ -298,19 +419,266 @@ describe('_getManifestResponseAsync', () => {
+        R/JfKcH1aL0yrJ/cvb0exwNyu7CwRoN/ree9SxGRqS0E31m3kZDTB8n2ZVWMFjoN
+        /zRrDy1m1JuWyzcmwfOKk8MDxJWkOwUplLdMi3pA39vVx88Rb/4qQ4BX+TNkuSmH
+        aoI01H5nPhIJROdrNOU0YfrPE15bl+N0Nq8MU/ygsTJstxNTck+CEmjYhT7gn/kT
+        kzwhP8wRAoGBAOwtoWYkvGWMmQ7nPJSrqyZZFg7TnCeIshtvrZsMl6VmxXnBAfCs
+        lcPBbX6XbHRyoVhZeWV0YbvVNqq6LVjto3pZjQdzLbr7ef+biciJU9cVT9RNVjcf
+        fraoa9vPPsLW4k4EpyXPSMPCTtYokGgqtDvQhiYw2SJKBu2CmG2RMiNXAoGBAM3G
+        Y4k6Fy7hbWGdZOacYYFtcoSkfiQKaoUc8/AOkY5KpvDWYcXsjs/ZX7mT5w9RPL05
+        YE9+vp0bStyWG4sCjvJZzaUIxkQLMFSLr6JIah422x97OyT6YwC/ebClksdRBXG6
+        mN20YghJSrSU56bvPngJWbnkaoToqIP9vz0Y3Kr5AoGBAKINjTkw67QiDMOVFpRM
+        VlsxtVjnCbZ2rP72WgCRW1w6cDVioZoqUXjIOAEslVYxcVPV0N40RPtFCQWMSaFD
+        LXC+EW6TMUWjV8pzRs86qzRhgy0BA68/BQr/9UIVqdK27wDpYGcpG5x2rvFzU+Kl
+        qpWANwvtaI+Gi8pcWYqUnfpxAoGAOnuB2oR6BkbnUXDsNmaHh8NVYsFlYFTdhUt/
+        QODDGY+59oTXcOsUg5oiDJoudzn9a0IdTzqUbg5NrnezCEUH33UKbzcZMkmDyOOS
+        XSr7je7FRCM/4fiPnqzMpxnHp4Ita+8+dvRHUFeUNdKEK/ue43q015JFe9Sr4CeD
+        7JdeT/ECgYEAmmkGP0SBXG97loeWrTnelLIdxExpEeQ3PHWkJBjg/fRNa8Qack+3
+        o0G+LL1P0gRD0XZcccIaJ38QIoL63xx7yCCRlw821gB9SrNJau8+rl6LE3lF22RG
+        KM04JH2KT9s6MrNy70DdYRjuh2lQeK6kbtur8XYvW/fEYVo1wX/PxDg=
+        -----END RSA PRIVATE KEY-----`,
+    });
+
+    const middleware = createMiddleware({
+      updates: {
+        codeSigningCertificate: 'keys/cert.pem',
+        codeSigningMetadata: {
+          keyid: 'testkeyid',
+          alg: 'rsa-v1_5-sha256',
+        },
+      },
+    });
+
+    const results = await middleware._getManifestResponseAsync({
+      explicitlyPrefersMultipartMixed: true,
+      platform: 'android',
+      acceptSignature: false,
+      expectSignature: 'sig, keyid="testkeyid", alg="rsa-v1_5-sha256"',
+      hostname: 'localhost',
+    });
+    expect(results.version).toBe('45.0.0');
+
+    const { body, headers } = await getMultipartPartAsync('manifest', results);
+    expect(headers.get('expo-signature')).toContain('keyid="expo-go"');
+
+    expect(JSON.parse(body)).toEqual({
+      id: expect.any(String),
+      createdAt: expect.any(String),
+      runtimeVersion: '45.0.0',
+      launchAsset: {
+        key: 'bundle',
+        contentType: 'application/javascript',
+        url: 'https://localhost:8081/bundle.js',
+      },
+      assets: [],
+      metadata: {},
+      extra: {
+        eas: {
+          projectId: 'projectId',
+        },
+        expoClient: expect.anything(),
+        expoGo: {},
+        scopeKey: expect.not.stringMatching(/@anonymous\/.*/),
+      },
+    });
+
+    const certificateChainMultipartPart = await getMultipartPartAsync('certificate_chain', results);
+    expect(certificateChainMultipartPart).toBeNull();
+  });
+
+  it('returns a code signed manifest with developers own key when requested and private-key-path arg is supplied', async () => {
+    vol.fromJSON({
+      'keys/cert.pem': `-----BEGIN CERTIFICATE-----
```

Comment:
Maybe move some of these into external fixture files to keep the test file scannable.

Reply:
Could even be a single `fixture/certificates.ts` which exports various string fixtures.

Reply:
Could even be a single `fixture/certificates.ts` which exports various string fixtures.

Reply:
I would like to but since some of these need to be used in `jest.mock` it seems jest prevents out of scope variables. I could move the few that are in the code (like the one in this comment), but it seemed to me like this would potentially be more confusing for a reader.

Reply:
If the variables are named `mockXXX` they can be accessed from within `jest.mock()`. This doesn't guarantee the tests will initialize correctly if the module mocks access the variables synchronously, but that's not the case in this code. Something like this may work, though I'm not sure if Jest's escape hatch applies to imports:
```
import {
  mockProjectDevelopmentCertificate,
  mockSelfSignedCertificate,
  mockSelfSignedCertificatePrivateKey,
} from './fixture/certificates';
```

The other way to move the certificates out of this file is to define `__mocks__/@expo/code-signing-certificates.ts` and put the whole mock implementation in there.